src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -0,0 +1,195 @@
+   * @param optimizer Desired optimizer.
+   */
+  template <typename OptimizerType>
+  LinearSVM(const MatType& data,
+            const arma::Row<size_t>& labels,
+            const size_t numClasses = 2,
+            const double lambda = 0.0001,
+            OptimizerType optimizer = OptimizerType());
+
+  /**
+   * Initialize the SoftmaxRegression without performing training.  Default
+   * value of lambda is 0.0001.  Be sure to use Train() before calling
+   * Classify() or ComputeAccuracy(), otherwise the results may be meaningless.
+   *
+   * @param inputSize Size of the input feature vector.
+   * @param numClasses Number of classes for classification.
+   * @param fitIntercept add intercept term or not.
+   */
+  LinearSVM(const size_t inputSize = 0,
+            const size_t numClasses = 0);
+
+  /**
+   * Classify the given points, returning the predicted labels for each point.
+   * The function calculates the probabilities for every class, given a data
+   * point. It then chooses the class which has the highest probability among
+   * all.
+   *
+   * @param data Set of points to classify.
+   * @param labels Predicted labels for each point.
+   */
+  void Classify(const MatType& data,
+                arma::Row<size_t>& labels) const;
+
+  /**
+   * Classify the given points, returning class scores and predicted
+   * class label for each point.
+   * The function calculates the scores for every class, given a data
+   * point. It then chooses the class which has the highest probability among
+   * all.
+   *
+   * @param data Matrix of data points to be classified.
+   * @param labels Predicted labels for each point.
+   * @param scores Class probabilities for each point.
+   */
+  void Classify(const MatType& data,
+                arma::Row<size_t>& labels,
+                arma::mat& scores) const;
+
+  /**
+   * Classify the given points, returning class scores for each point.
+   *
+   * @param data Matrix of data points to be classified.
+   * @param scores Class scores for each point.
+   */
+  void Classify(const MatType& data,
+                arma::mat& scores) const;
+
+  /**
+   * Classify the given point. The predicted class label is returned.
+   * The function calculates the scores for every class, given the point.
+   * It then chooses the class which has the highest probability among all.
+   *
+   * @param point Point to be classified.
+   * @return Predicted class label of the point.
+   */
+  template<typename VecType>
+  size_t Classify(const VecType& point) const;
+
+  /**
+   * Computes accuracy of the learned model given the feature data and the
+   * labels associated with each data point. Predictions are made using the
+   * provided data and are compared with the actual labels.
+   *
+   * @param testData Matrix of data points using which predictions are made.
+   * @param testLabels Vector of labels associated with the data.
+   * @return Accuracy of the model.
```

Comment:
Can you point out whether this will be between 0 and 1 or 0 and 100?  (I think 0 and 1 would be better, but I don't have a strong opinion.)

Reply:
`0 and 100` is fine, but can you mark that in the documentation?  `LogisticRegression::ComputeAccuracy()` also functions the same way.